sympy/sets/sets.py

```diff
@@ -969,6 +969,8 @@ def _contains(self, other):
 
         if self.start is S.NegativeInfinity and self.end is S.Infinity:
             if not other.is_real is None:
+                if other is S.NegativeInfinity or other is S.Infinity:
+                    return false
```

Comment:
I think, we can check for `other` belonging to extended reals without even checking `self` for the same.